docs/source/cpp/api.rst

```diff
@@ -45,3 +45,4 @@ API Reference
    api/flightsql
    api/filesystem
    api/dataset
+   api/extension
```

Comment:
How about moving this after `datatype` and `array` because this is related to `datatype` and `array`?

```diff
diff --git a/docs/source/cpp/api.rst b/docs/source/cpp/api.rst
index 27b78b57ff..b58a7b7aeb 100644
--- a/docs/source/cpp/api.rst
+++ b/docs/source/cpp/api.rst
@@ -27,6 +27,7 @@ API Reference
    api/thread
    api/datatype
    api/array
+   api/extension
    api/scalar
    api/builder
    api/table
```

Reply:
Thanks! Fixed.